src/ResourceManager/IotHub/Commands.IotHub/IotHub/AddAzureRmIotHubCertificate.cs

```diff
@@ -0,0 +1,107 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Management.IotHub
+{
+    using System;
+    using System.Globalization;
+    using System.IO;
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.Management.IotHub.Common;
+    using Microsoft.Azure.Commands.Management.IotHub.Models;
+    using Microsoft.Azure.Management.IotHub;
+    using Microsoft.Azure.Management.IotHub.Models;
+
+    [Cmdlet(VerbsCommon.Add, "AzureRmIotHubCertificate", SupportsShouldProcess = true)]
+    [OutputType(typeof(PSCertificateDescription))]
+    public class AddAzureRmIotHubCertificate : IotHubBaseCmdlet
+    {
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Name of the Resource Group")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Name of the Iot Hub")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            Position = 2,
+            Mandatory = true,
+            HelpMessage = "Name of the Certificate")]
+        [ValidateNotNullOrEmpty]
+        public string CertificateName { get; set; }
+
+        [Parameter(
+            Position = 3,
+            Mandatory = true,
+            HelpMessage = "base-64 representation of X509 certificate .cer file or .pem file path.")]
+        [ValidateNotNullOrEmpty]
+        public string Path { get; set; }
+
+        [Parameter(
+            Position = 4,
+            Mandatory = false,
+            HelpMessage = "Etag of the Certificate")]
+        [ValidateNotNullOrEmpty]
+        public string Etag { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (ShouldProcess(CertificateName, Properties.Resources.AddIotHubCertificate))
+            {
+                string certificate = string.Empty;
+                FileInfo fileInfo = new FileInfo(this.Path);
+                switch (fileInfo.Extension.ToLower(CultureInfo.InvariantCulture))
```

Comment:
Switching based on file extension can be dangerous since a user can change the file extension of a certificate file. Have a look at how I did this for python. I'd suggest trying to decode the contents as utf-8, if that fails you know you should probably base-64 encode the contents. See this: https://github.com/Azure/azure-cli/pull/4804/files#diff-f1d6ba6f7c8eff1c2280a7394f40ffd4

Reply:
Done